pkg/mountutil/mountutil.go

```diff
@@ -161,5 +158,150 @@ func ProcessFlagV(s string, volStore volumestore.VolumeStore, createDir bool) (*
 		}
 		res.Mount.Options = strutil.DedupeStrSlice(append(res.Mount.Options, unpriv...))
 	}
-	return &res, nil
+
+	return res, nil
+}
+
+func handleVolumeToMount(source string, volStore volumestore.VolumeStore, createDir bool) (volumeSpec, error) {
+	var res volumeSpec
+
+	res.Type = Bind
+	res.Source = source
+
+	// Handle named volumes
+	if isNamedVolume(source) {
+		return handleNamedVolumes(source, volStore)
+	}
+
+	// Handle named pipes
+	if isNamedPipe(source) {
+		res.Type = Npipe
+		return res, nil
+	}
+
+	// Handle relative paths
+	if !filepath.IsAbs(source) {
+		log.L.Warnf("expected an absolute path, got a relative path %q (allowed for nerdctl, but disallowed for Docker, so unrecommended)", source)
+		absPath, err := filepath.Abs(source)
+		if err != nil {
+			return res, fmt.Errorf("failed to get the absolute path of %q: %w", source, err)
+		}
+		res.Source = absPath
+	}
+
+	// Create dir if it does not exist
+	if err := createDirOnHostIfNotExist(source, createDir); err != nil {
```

Comment:
nit: perhaps `createDirOnHost` would do? The _if not exist_ part can be implicit IMO.